### DISCUSSION TITLE
Telemetry - expose data for internal / API use.

### DISCUSSION THREAD

**Discussion by thruflo** [electric-sql/electric#1073] [Feature requests]
To paste from discussion in Discord:

> Hey I'm looking for pointers on how best to monitor what changes are being transmitted to/from clients. I see it's protobuf going over the websocket so that wasn't much help, maybe there's a good spot in the elixir app to add some debug logging? (or even better, enable if it already exists)

> We can have a dashboard that shows MAU, current users connected, what changes are synced etc. For example Firebase and Supabase dashboards.

> There’s telemetry data built in: https://electric-sql.com/docs/reference/telemetry - I was just checking docs to see if there’s an env var to configure the URL. If there isn’t there should be.

> What I can understand is this data collected and used by  the electric team. If we can even configure the URL then also the user would have to create a service to handle this. What I was talking about is that we can create a dashboard that queries the electric service using endpoints. To show the usage etc.

> Yup, it’s based on standard Elixir telemetry. And it just posts JSON. So it may be something to work with.

> Perhaps we could simply have an option to persist the telemetry report to the source Postgres. Then some analytics functions could just query across that. https://github.com/electric-sql/electric/blob/main/components/electric/lib/electric/telemetry/call_home_reporter.ex#L28

> A dashboard would really helpful. In the shorter term though being able to get the telemetry into something like datadog would be really helpful. Bridging the elixir telemetry to open telemetry seems like a good approach.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
